js/app.js

```diff
@@ -37,45 +50,70 @@ function handleAddRateForm(event) {
     .value.trim()
     .toUpperCase();
 
-  // Get the value of the exchange rate from the input field and convert it to a float
+  // Get the value of the exchange rate from the input field
   const exchangeRate = parseFloat(
     document.getElementById("exchange-rate").value
   );
 
-  // Find the rate object for the base currency, or create a new one if it doesn't exist
-  let rateObject = currencyRates.find((rate) => rate.base === baseCurrency);
-
-  if (!rateObject) {
-    // If no rate object is found for the base currency
-    rateObject = {
-      // Create a new rate object
-      timestamp: Date.now(), // Current timestamp
-      base: baseCurrency, // Set the base currency
-      date: new Date().toISOString().split("T")[0], // Current date in YYYY-MM-DD format
-      rates: {}, // Initialize an empty rates object
-    };
-    currencyRates.push(rateObject); // Add the new rate object to the currencyRates array
+  // Find an existing rate if it exists
+  const existingRate = currencyRates.find(
```

Comment:
You could also benefit from a search function here. See my other comments about it: